Svc/ComQueue/ComQueue.cpp

```diff
@@ -154,12 +159,21 @@ void ComQueue::comStatusIn_handler(const FwIndexType portNum, Fw::Success& condi
         // On success, the queue should be processed. On failure, the component should still wait.
         case WAITING:
             if (condition.e == Fw::Success::SUCCESS) {
+#if FW_COM_BUFFER_RETRY_ON_FAILURE
+                this->m_hasRetried = false;
+#endif
                 this->m_state = READY;
                 this->processQueue();
                 // A message may or may not be sent. Thus, READY or WAITING are acceptable final states.
                 FW_ASSERT((this->m_state == WAITING || this->m_state == READY),
                           static_cast<FwAssertArgType>(this->m_state));
             } else {
+#if FW_COM_BUFFER_RETRY_ON_FAILURE
```

Comment:
Our coding standards discourage this practice. An alternate solution would be to use a `.configure(bool does_retry)` method that can set a member variable into the component as to whether it should retry or not. Then add this to your conditionals.

Reply:
Thank you for the review. I'll check this out.

Reply:
Sure thing. I'll do that.

Reply:
Awesome. Thanks a lot for contributing! 🚀 
This is a much needed feature.

Reply:
@valdaarhun we already have a configure method, so following the suggested approach I'd recommend you add a bool arg to it e.g.

```
void ComQueue::configure(..., bool retry_on_failure)
```